### ISSUE TITLE
Inconsistency between `map` and `select`; proposal for a new `filter` function

### ISSUE THREAD

**Issue by rbren** [mikefarah/yq#1586]
Apologies if this is a dupe, but I didn't see anything.

**Please describe your feature request.**
I'd like to use `map` and `select` the same way I use `map` and `filter` in JavaScript--to turn arrays into arrays. `map` does this very well already.

Consider this example:
```bash
arr="
- amt: 1
- amt: 2
- amt: 3
"
echo "${arr}" | yq e '. | map(.amt + 1)' -
echo "---"
echo "${arr}" | yq e '.[] | select(.amt > 1)' -
```

This outputs:
```yaml
- 2
- 3
- 4
---
2
3
```
Note that `map` outputs a valid YAML array, while `select` outputs raw lines of text.

**Describe the solution you'd like**
```bash
arr="
- amt: 1
- amt: 2
- amt: 3
"
echo "${arr}" | yq e '. | filter(.amt > 1)' -
```
which would output
```bash
- 2
- 3
```

Note that this makes chaining really nice! E.g.
```bash
echo "${arr}" | yq e '. | filter(.amt > 1) | map(.amt * 2) | filter(. > 3)' -
```

**Describe alternatives you've considered**
It's possible this has been solved by another function I'm not seeing.

**Additional context**

**Comment by rbren** [comment]
Also--I'm happy to implement this if you think it's a good idea!

**Comment by mikefarah** [comment]
Yeah nice - I think that would work well. I'd be very happy to take a PR :)

**Comment by mikefarah** [comment]
Released in 4.32.1 - Thanks again @rbren  :)
